delfin/drivers/hpe/hpe_3par/component_handler.py

```diff
@@ -417,3 +417,242 @@ def parse_speed(self, speed_value):
+        result = {}
+        if volume_groups:
+            volumes_map = self.ssh_handler.get_resources_ids(
+                self.ssh_handler.HPE3PAR_COMMAND_SHOWVV,
+                consts.HOST_OR_VV_PATTERN)
+            for volume_group in volume_groups:
+                volume_members = volume_group.get('members')
+                volume_ids = []
+                if volumes_map:
+                    for volume_name in (volume_members or []):
+                        volume_id = volumes_map.get(volume_name)
+                        if volume_id:
+                            volume_ids.append(volume_id)
+                volume_group_model = {
+                    "name": volume_group.get('name'),
+                    "description": volume_group.get('comment'),
+                    "storage_id": storage_id,
+                    "native_volume_group_id": volume_group.get('id'),
+                    "volumes": ','.join(volume_ids)
+                }
+                volume_group_list.append(volume_group_model)
+            volume_group_relation_list = []
+            for volume_group in volume_group_list:
+                volumes = volume_group.pop('volumes', None)
+                if not volumes:
+                    continue
+                volumes = volumes.split(',')
+
+                for volume in volumes:
+                    volume_group_relation = {
+                        'storage_id': storage_id,
+                        'native_volume_group_id':
+                            volume_group.get('native_volume_group_id'),
+                        'native_volume_id': volume}
+                    volume_group_relation_list.append(volume_group_relation)
+
+            result = {
+                'volume_groups': volume_group_list,
+                'vol_grp_vol_rels': volume_group_relation_list
+            }
+        return result
+
+    def list_masking_views(self, storage_id):
+        views = self.ssh_handler.list_masking_views()
+        views_list = []
+        if views:
+            hosts_map = self.ssh_handler.get_resources_ids(
+                self.ssh_handler.HPE3PAR_COMMAND_SHOWHOST_D,
+                consts.HOST_OR_VV_PATTERN)
+            hosts_group_map = self.ssh_handler.get_resources_ids(
+                self.ssh_handler.HPE3PAR_COMMAND_SHOWHOSTSET_D,
+                consts.HOST_OR_VV_PATTERN)
+            volumes_map = self.ssh_handler.get_resources_ids(
+                self.ssh_handler.HPE3PAR_COMMAND_SHOWVV,
+                consts.HOST_OR_VV_PATTERN)
+            volumes_group_map = self.ssh_handler.get_resources_ids(
+                self.ssh_handler.HPE3PAR_COMMAND_SHOWVVSET_D,
+                consts.HOST_OR_VV_PATTERN)
+            for view in views:
+                vv_name = view.get('vvname')
+                host_name = view.get('hostname')
+                if vv_name and host_name:
+                    port = view.get('port', '').replace('-', '')
+                    id = view.get('lun')
```

Comment:
`id` is a built-in name, please use another name instead.

Reply:
Modified